### PR TITLE
Proper handling of Notabene error format 

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Handle identity provider sending `error` instead of `code_uri`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/background/identity-issuance.ts [hunk 1]
```diff
@@ -145,20 +172,20 @@ async function startIdentityIssuance({
     }
 
     try {
+        chrome.webRequest.onBeforeRedirect.addListener(
+            (response) => {
+                if (!response.redirectUrl.includes(redirectUri)) {
+                    launchExternalIssuance(response.redirectUrl);
+                }
+            },
+            { urls: ['<all_urls>'], tabId: -1 }
```
Comment (soerenbf) [inline packages/browser-wallet/src/background/identity-issuance.ts:181]: Maybe make a const?

```suggestion
            { urls: ['<all_urls>'], tabId: IDP_FETCH_TAB_ID }
```

File: packages/browser-wallet/src/background/identity-issuance.ts [hunk 2]
```diff
@@ -60,20 +61,46 @@ export function addIdpListeners() {
...
     });
 
-    const handleIdpRequest = (redirectUrl: string, tabId: number) => {
-        chrome.tabs.remove(tabId);
-        sessionIdpTab.remove();
+    const handleIdpRequest = (redirectUrl: string, tabId?: number) => {
+        if (tabId !== undefined) {
+            chrome.tabs.remove(tabId);
+            sessionIdpTab.remove();
+        }
```
Comment (soerenbf) [inline packages/browser-wallet/src/background/identity-issuance.ts:68]: Do we want this to run for error requests with tabID -1?

Comment (shjortConcordium) [reply packages/browser-wallet/src/background/identity-issuance.ts:68]: As far as I understand the "protocol", the error request only happens before we set it, so the tabId will be undefined. But i've changed the condition to be more obvious.

File: packages/browser-wallet/src/background/identity-issuance.ts [hunk 3]
```diff
@@ -145,20 +172,20 @@ async function startIdentityIssuance({
...
 
     try {
+        chrome.webRequest.onBeforeRedirect.addListener(
+            (response) => {
+                if (!response.redirectUrl.includes(redirectUri)) {
+                    launchExternalIssuance(response.redirectUrl);
+                }
+            },
+            { urls: ['<all_urls>'], tabId: -1 }
+        );
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/background/identity-issuance.ts]: I didn't need it actually. Removed it.

File: packages/browser-wallet/src/background/identity-issuance.ts [hunk 4]
```diff
@@ -60,20 +61,46 @@ export function addIdpListeners() {
...
 
     chrome.webRequest.onBeforeRequest.addListener(
         (details) => {
             sessionIdpTab.get().then((idpTabId) => {
-                if (details.url.includes(redirectUri) && details.tabId === idpTabId) {
+                if (
+                    (details.url.includes(`${redirectUri}#${codeUriKey}`) ||
+                        details.url.includes(`${redirectUri}#${errorKey}`)) &&
+                    (details.tabId === idpTabId || details.tabId === -1)
+                ) {
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/background/identity-issuance.ts]: I like it.

File: packages/browser-wallet/src/background/identity-issuance.ts (hunk 1 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/background/identity-issuance.ts]: Removed it.

**Review by soerenbf — APPROVED** [review]

**Pull request merged**